builtin/common/information_formspecs.lua

```diff
@@ -110,6 +119,116 @@ local function build_privs_formspec(name)
 end
 
 
+-- Checks whether the mod with the given name is a separate mod or a part of a modpack
+local function is_mod_sep_or_part_of_mp(modname)
+	local modpath = core.get_modpath(modname)
+
+	local ptn = DIR_DELIM .. "[^" .. DIR_DELIM .. "]+"
+	local sep_mod_ptn = ptn .. "$"
+	local mp_ptn = ptn:rep(2) .. "$"
+	local mods_ptn = DIR_DELIM .. "mods"
+	local worldmods_ptn = DIR_DELIM .. "worldmods"
+
+	if modpath:match(mods_ptn .. mp_ptn) or modpath:match(worldmods_ptn .. mp_ptn) then
+		return "modpack", modpath:match(DIR_DELIM .. "([^" .. DIR_DELIM .. "]+)" .. DIR_DELIM .. "[^" .. DIR_DELIM .. "]+$")
+	elseif modpath:match(mods_ptn .. sep_mod_ptn) or modpath:match(worldmods_ptn .. sep_mod_ptn) then
+		return "sep_mod"
+	end
+	--[[local find_ptn = "[^" .. DIR_DELIM .. "]*.$"
+	local modpart = modpath:match(find_ptn)
+	local mp_part = modpath:sub(1, modpath:len()-modpart:len()-1):match(find_ptn)
+
+	if mp_part ~= "mods" then   -- this is inside of the modpack!
+		return mp_part, modpart
+	else                        -- this is a just separate mod!
+		return modpart
+	end]]
+end
+
+
+-- Returns a list with the modpack`s mods
+local function get_modpack_mods(mp_name)
+	local modnames = core.get_modnames()
+
+	local modlist = {}
+	for i, mname in ipairs(modnames) do
+		local mpath = core.get_modpath(mname)
+
+		-- If the mod is saving in '/games/<game_name>/mods/<modpack_name>/' or in '/mods/<modpack_name>/'
+		local match1 = mpath:match(DIR_DELIM .. "mods" .. DIR_DELIM .. mp_name .. DIR_DELIM .. mname .. "$")
+		-- else if the mod is saving in '/worlds/<world_name>/worldmods/<modpack_name>'
+		local match2 = mpath:match(DIR_DELIM .. "worldmods" .. DIR_DELIM .. mp_name .. DIR_DELIM .. mname .. "$")
+
+		if match1 or match2 then
+			table.insert(modlist, mname)
+		end
+	end
+
+	return modlist
+end
+
+
+-- MODS FORMSPEC
+
+local mods_formspec_cache = ""
+
+function build_mods_formspec()
+	if mods_formspec_cache:len() > 0 then
+		return mods_formspec_cache
+	end
+
+	local modslist = core.get_modnames()
+	local mps_and_smods = {}
+
+	local rows = {}
+	rows[1] = "#FFF,0," .. F(S("Mod or Modpack Name"))
+
+	local function find_modpack(mp_name)
+		for i, t in ipairs(mps_and_smods) do
+			if t[1] == mp_name and t[2] == "modpack" then
+				return true
+			end
+		end
+
+		return false
+	end
+
+	for i = 1, #modslist do
+		local mod_t, mp_name = is_mod_sep_or_part_of_mp(modslist[i])
+
+		if mod_t == "modpack" and mp_name then
+			if not find_modpack(mp_name) then
+				mps_and_smods[#mps_and_smods+1] = {mp_name, "modpack"}
```

Comment:
Giving names to object properties helps readability, consider the following example:
```
table.insert(mps_and_smods, { name = mp_name, type = "modpack" })
...
if mps_and_smods[i].type == "modpack" then ... end
```